tests/test_models.py

```diff
@@ -229,6 +257,32 @@ def test_scoped_named(self):
         self.session.commit()
         self.assertEqual(d4.permissions(user=u), set([]))
 
+        # test insert in BaseScopedNameMixin's upsert
+        d4 = ScopedNamedDocument.upsert(c1, u'hello4', title=u'Hello 4', content=u'scoped named doc')
+        self.session.commit()
+        d4_persisted = ScopedNamedDocument.get(c1, u'hello4')
+        self.assertEqual(d4_persisted, d4)
+        self.assertEqual(d4_persisted.content, u'scoped named doc')
+
+        # test update in BaseScopedNameMixin's upsert
+        d5 = ScopedNamedDocument.upsert(c1, u'hello4', container=c2, title=u'Hello5', content=u'scoped named doc')
+        d5.make_name()
+        self.session.commit()
+        d5_persisted = ScopedNamedDocument.get(c2, u'hello5')
+        self.assertEqual(d5_persisted, d5)
+        self.assertEqual(d5_persisted.content, u'scoped named doc')
+
+        with self.assertRaises(TypeError) as insert_error:
+            ScopedNamedDocument.upsert(c1, u'invalid1', title=u'Invalid1', non_existent_field=u"I don't belong here.")
+        self.assertEqual(TypeError, insert_error.expected)
+
+        with self.assertRaises(TypeError) as update_error:
```

Comment:
This `with` should be two lines down. You want to confirm the valid field gets set without error.